### ISSUE TITLE
Unable to create vpc intefaces

### ISSUE THREAD

**Issue by tias77** [CiscoDevNet/ansible-dcnm#43]
DCNM v11.5.1

Collection        Version
----------------- -------
ansible.netcommon 1.4.1  
cisco.dcnm        1.0.0  

```
---

- hosts: dcnm_controllers
  gather_facts: false
  connection: ansible.netcommon.httpapi

  tasks:
    - name: Create vPC interfaces
      cisco.dcnm.dcnm_interface: &vpc_merge
        fabric: f
        state: merged                         # only choose from [merged, replaced, deleted, overridden, query]
        config:
          - name: vpc1
            type: vpc
            switch:
              - "le1"
              - "le2"
            deploy: true                       # choose from [true, false]
            profile:
              admin_state: true                ## choose from [true, false]
              mode: trunk                      # choose from [trunk, access]
              peer1_pcid: 1                    # choose between [Min:1, Max:4096], if not given, will be VPC port-id
              peer2_pcid: 1                    # choose between [Min:1, Max:4096], if not given, will be VPC port-id
              peer1_members:                   ## member interfaces on peer 1
                - e1/1
              peer2_members:                   ## member interfaces on peer 2
                - e1/1
              pc_mode: 'active'                ## choose from ['on', 'active', 'passive']
              bpdu_guard: true                 ## choose from [true, false, 'no']
              port_type_fast: true             ## choose from [true, false]
              mtu: jumbo                       ## choose from [default, jumbo]
              peer1_allowed_vlans: none        ## choose from [none, all, vlan range]
              peer2_allowed_vlans: none        ## choose from [none, all, vlan range]
              peer1_description: "VPC acting as trunk peer1 - modified"
              peer2_description: "VPC acting as trunk peer2 - modified"
```

generates the following dry run:
```
changed: [dcnm] => {
    "changed": true,
    "diff": [
        {
            "deleted": [],
            "deploy": [
                {
                    "ifName": "vPC1",
                    "serialNumber": "serials"
                }
            ],
            "merged": [
                {
                    "interfaces": [
                        {
                            "fabricName": "f",
                            "ifName": "vPC1",
                            "interfaceType": "INTERFACE_VPC",
                            "nvPairs": {
                                "ADMIN_STATE": "true",
                                "BPDUGUARD_ENABLED": "true",
                                "INTF_NAME": "vPC1",
                                "MTU": "jumbo",
                                "PC_MODE": "active",
                                "PEER1_ALLOWED_VLANS": "none",
                                "PEER1_MEMBER_INTERFACES": "e1/1",
                                "PEER1_PCID": "1",
                                "PEER1_PO_CONF": "",
                                "PEER1_PO_DESC": "VPC acting as trunk peer1 - modified",
                                "PEER2_ALLOWED_VLANS": "none",
                                "PEER2_MEMBER_INTERFACES": "e1/1",
                                "PEER2_PCID": "1",
                                "PEER2_PO_CONF": "",
                                "PEER2_PO_DESC": "VPC acting as trunk peer2 - modified",
                                "PORTTYPE_FAST_ENABLED": "true"
                            },
                            "serialNumber": "serials"
                        }
                    ],
                    "policy": "int_vpc_trunk_host_11_1",
                    "skipResourceCheck": "true"
                }
            ],
            "overridden": [],
            "query": [],
            "replaced": []
        }
    ],
    "invocation": {
        "module_args": {
            "config": [
                {
                    "deploy": true,
                    "name": "vpc1",
                    "profile": {
                        "admin_state": true,
                        "bpdu_guard": true,
                        "mode": "trunk",
                        "mtu": "jumbo",
                        "pc_mode": "active",
                        "peer1_allowed_vlans": "none",
                        "peer1_description": "VPC acting as trunk peer1 - modified",
                        "peer1_members": [
                            "e1/1"
                        ],
                        "peer1_pcid": 1,
                        "peer2_allowed_vlans": "none",
                        "peer2_description": "VPC acting as trunk peer2 - modified",
                        "peer2_members": [
                            "e1/1"
                        ],
                        "peer2_pcid": 1,
                        "port_type_fast": true
                    },
                    "switch": [
                        "le1",
                        "le2"
                    ],
                    "type": "vpc"
                }
            ],
            "fabric": "f",
            "state": "merged"
        }
    },
    "response": []
}
```

But when running it bails on interface type:
```
fatal: [dcnm]: FAILED! => {
    "changed": false,
    "invocation": {
        "module_args": {
            "config": [
                {
                    "deploy": true,
                    "name": "vpc1",
                    "profile": {
                        "admin_state": true,
                        "bpdu_guard": true,
                        "mode": "trunk",
                        "mtu": "jumbo",
                        "pc_mode": "active",
                        "peer1_allowed_vlans": "none",
                        "peer1_description": "VPC acting as trunk peer1 - modified",
                        "peer1_members": [
                            "e1/1"
                        ],
                        "peer1_pcid": 1,
                        "peer2_allowed_vlans": "none",
                        "peer2_description": "VPC acting as trunk peer2 - modified",
                        "peer2_members": [
                            "e1/1"
                        ],
                        "peer2_pcid": 1,
                        "port_type_fast": true
                    },
                    "switch": [
                        "le1",
                        "le2"
                    ],
                    "type": "vpc"
                }
            ],
            "fabric": "f",
            "state": "merged"
        }
    },
    "msg": {
        "DATA": [
            {
                "column": 0,
                "entity": "",
                "line": 0,
                "message": "Interface type is not valid",
                "reportItemType": "ERROR"
            }
        ],
        "MESSAGE": "Internal Server Error",
        "METHOD": "POST",
        "REQUEST_PATH": "https://dcnm:443/rest/globalInterface",
        "RETURN_CODE": 500
    }
}
```
rest_api_trace.log:
```
--------------------------------------------------------------------------------
2021.03.19 10:53:23.501  INFO  - REQUEST    [id=36454]
	POST        /rest/globalInterface
	request-id: 36454
	client-ip:  ip
	username:   
	size:       683 B   (683)
	upload time: 
--------------------------------------------------------------------------------
2021.03.19 10:53:23.536  INFO  - RESPONSE    [id=36454]
	POST        /rest/globalInterface
	request-id:  36454
	client-ip:   ip
	username:    admin
	took:        35 ms
	http code:   500
[ {
  "reportItemType" : "ERROR",
  "message" : "Interface type is not valid",
  "entity" : "",
  "line" : 0,
  "column" : 0
} ]
--------------------------------------------------------------------------------
```

**Comment by mikewiebe** [comment]
@tias77 Thanks for bringing this to our attention.  We are looking into the failure.

**Comment by mmudigon** [comment]
The issue is fixed in the develop branch with https://github.com/CiscoDevNet/ansible-dcnm/pull/31 request and will be in the next Ansible Galaxy Release

**Comment by tias77** [comment]
@mmudigon Great news, when do we expect to see it released?

**Comment by mikewiebe** [comment]
@tias77 Our goal is to publish the new version in the next couple of weeks

**Comment by jrragan** [comment]
Is there a date yet for when the next Ansible Galaxy Release containing this bug fix will be available? Thanks!

**Comment by mikewiebe** [comment]
@jrragan We are planning to publish this Friday 4/16.

**Comment by mikewiebe** [comment]
@jrragan and @tias77 We just finished publishing the latest release `1.1.0` to galaxy.  Please let us know if you encounter any more problems with this issue.  Thanks!

**Comment by mikewiebe** [comment]
This is fixed in the latest release `1.1.0`
